DOCUMENTATION.md

```diff
@@ -0,0 +1,305 @@
+# Reference API
+
+## HTML
+
+### Root Element
+
+impress.js requires a Root Element with id "impress". All the content of the presentation will be created inside that element. It is not recommended to manipulate any of the styles, attributes or classes that are created by impress.js inside the Root Element after initialization.
```

Comment:
> The id of root element doesn't have to be "impress" (but is "impress" by default). It is configurable in impress "constructor".
> For example:

I will edit that to consider those documented use cases.

> In theory there can be multiple root elements on a page with multiple instances of impress being initialized on them.
> But at the moment keyboard events will work on all of them at the same time.

Is multiple root elements on a page actually supported? If it is not explicitly commented in `index.html` shouldn't we just refrain from supporting it unless someone asks to do so?

People over time asked a lot for substeps, in this case if we implement substeps we might automatically fix the use cases for multiple instances of impress in the page and then we could manage to support it.

Does that make sense?